### PR TITLE
Connect-DbaInstance - Optimize code for StatementTimeout and ApplicationIntent 

### PR DESCRIPTION
**Pull request by andreasjordan**
As StatementTimeout is an integer variable with a default from an integer configuration - this can never be $null. So we always set this.
But to not copy connection context if not needed, added a test like for Database and NonPooledConnection. Then did that also with ApplicationIntent to have them all aligned.

### CODE REVIEW

File: functions/Connect-DbaInstance.ps1 [hunk 1]
```diff
@@ -605,19 +608,19 @@ function Connect-DbaInstance {
...
                     if ($Database -and $inputObject.ConnectionContext.CurrentDatabase -ne $Database) {
-                        Write-Message -Level Verbose -Message "Parameter Database passed in, and it's not the same as currently in ConnectionContext.CurrentDatabase, so we copy the connection context"
+                        Write-Message -Level Verbose -Message "Parameter Database passed in, and it's not the same as currently in ConnectionContext.CurrentDatabase, so we copy the connection context and set the CurrentDatabase"
                         $copyContext = $true
                     }
-                    if ($ApplicationIntent) {
-                        Write-Message -Level Verbose -Message "Parameter ApplicationIntent passed in, so we copy the connection context and set the ApplicationIntent"
+                    if ($ApplicationIntent -and $inputObject.ConnectionContext.ApplicationIntent -ne $ApplicationIntent) {
+                        Write-Message -Level Verbose -Message "Parameter ApplicationIntent passed in, and it's not the same as currently in ConnectionContext.ApplicationIntent, so we copy the connection context and set the ApplicationIntent"
```
Comment (wsmelton) [inline functions/Connect-DbaInstance.ps1:615]:
```suggestion
                        Write-Message -Level Verbose -Message "ApplicationIntent provided. Does not match ConnectionContext.ApplicationIntent, copying ConnectionContext and setting the ApplicationIntent"
```

File: functions/Connect-DbaInstance.ps1 [hunk 2]
```diff
@@ -605,19 +608,19 @@ function Connect-DbaInstance {
...
                         $copyContext = $true
                     }
                     if ($NonPooledConnection -and -not $inputObject.ConnectionContext.NonPooledConnection) {
                         Write-Message -Level Verbose -Message "Parameter NonPooledConnection passed in and we currently have a pooled connection, so we copy the connection context and set NonPooledConnection"
                         $copyContext = $true
                     }
-                    if (Test-Bound -Parameter StatementTimeout) {
-                        Write-Message -Level Verbose -Message "Parameter StatementTimeout passed in, so we copy the connection context and set the StatementTimeout"
+                    if (Test-Bound -Parameter StatementTimeout -and $inputObject.ConnectionContext.StatementTimeout -ne $StatementTimeout) {
+                        Write-Message -Level Verbose -Message "Parameter StatementTimeout passed in, and it's not the same as currently in ConnectionContext.StatementTimeout, so we copy the connection context and set the StatementTimeout"
```
Comment (wsmelton) [inline functions/Connect-DbaInstance.ps1:623]:
```suggestion
                        Write-Message -Level Verbose -Message "StatementTimeout provided. Does not match ConnectionContext.StatementTimeout, copying ConnectionContext and setting the StatementTimeout"
```

File: functions/Connect-DbaInstance.ps1 [hunk 3]
```diff
@@ -605,19 +608,19 @@ function Connect-DbaInstance {
                     # We do not test for SqlCredential as this would change the behavior compared to the legacy code path
                     $copyContext = $false
                     if ($Database -and $inputObject.ConnectionContext.CurrentDatabase -ne $Database) {
-                        Write-Message -Level Verbose -Message "Parameter Database passed in, and it's not the same as currently in ConnectionContext.CurrentDatabase, so we copy the connection context"
+                        Write-Message -Level Verbose -Message "Parameter Database passed in, and it's not the same as currently in ConnectionContext.CurrentDatabase, so we copy the connection context and set the CurrentDatabase"
```
Comment (wsmelton) [inline functions/Connect-DbaInstance.ps1:611]:
```suggestion
                        Write-Message -Level Verbose -Message "Database provided. Does not match ConnectionContext.CurrentDatabase, copying ConnectionContext and setting the CurrentDatabase"
```

**Comment by potatoqualitee** [comment]
All tests pass and Shawn's changes have been addressed 👍🏼  Merging

**Pull request merged**